### PR TITLE
Move variantOf field assignment from Node API to Content API #9957

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

File: modules/core/core-api/src/main/java/com/enonic/xp/content/DuplicateContentParams.java [hunk 1]
```diff
@@ -94,6 +118,12 @@ public int hashCode()
 
         private Boolean includeChildren = true;
 
+        private boolean variant = false;
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/content/DuplicateContentParams.java:121]:
```suggestion
        private boolean variant;
```

File: modules/core/core-api/src/main/java/com/enonic/xp/content/DuplicateContentParams.java [hunk 2]
```diff
@@ -128,9 +158,31 @@ public Builder includeChildren( final Boolean includeChildren )
...
+        public Builder parent( final ContentPath parent )
+        {
+            this.parent = parent;
+            return this;
+        }
+
         public DuplicateContentParams build()
         {
             Preconditions.checkNotNull( this.contentId, "Content id cannot be null" );
+            if ( this.variant && this.includeChildren )
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/content/DuplicateContentParams.java:182]: Preconditions.checkArgument

**Review by rymsha — APPROVED** [review]

File: modules/core/core-api/src/main/java/com/enonic/xp/content/DuplicateContentParams.java [hunk 3]
```diff
@@ -124,13 +156,33 @@ public Builder duplicateContentListener( DuplicateContentListener duplicateConte
 
         public Builder includeChildren( final Boolean includeChildren )
         {
-            this.includeChildren = includeChildren;
+            this.includeChildren = Objects.requireNonNullElse( includeChildren, true );
+            return this;
+        }
+
+        public Builder variant( final Boolean variant )
```
Comment (vbradnitski) [inline modules/core/core-api/src/main/java/com/enonic/xp/content/DuplicateContentParams.java:163]: can't we use simple 'boolean' here?

Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/content/DuplicateContentParams.java:163]:
```suggestion
        public Builder variant( final boolean variant )
```

**Review by vbradnitski — CHANGES_REQUESTED** [review]

**Review by rymsha — CHANGES_REQUESTED** [review]

**Review by vbradnitski — APPROVED** [review]

**Review by rymsha — APPROVED** [review]

**Pull request merged**